core/templates/services/http.service.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview HTTP Service for exclusively build HTTP Requests.
+ */
+
+import { HttpHeaders, HttpParams, HttpClient } from
+  '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+interface HttpOptions {
+  params: HttpParams,
+  headers: HttpHeaders
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpService {
+  constructor(private httpClient: HttpClient) {}
+
+  /* Used when we want to exclusively read data; no modifications. */
+  get<T>(url: string, options?: HttpOptions): Promise<T> {
+    return this.httpClient.get<T>(url, options).toPromise();
+  }
+
+  /* Used when we want to fully replace the value of a resource. */
+  put<T>(url: string, options?: HttpOptions): Promise<T> {
+    return this.httpClient.put<T>(url, options).toPromise();
+  }
+
+  /**
+   *Used when, instead of providing a completely new version of a resource,
```

Comment:
Add space after * on this line and the next.

Reply:
Ok